pandas/core/arrays/datetimes.py

```diff
@@ -1647,6 +1649,19 @@ def maybe_convert_dtype(data, copy):
         raise TypeError("Passing PeriodDtype data is invalid.  "
                         "Use `data.to_timestamp()` instead")
 
+    elif is_categorical_dtype(data):
+        # TODO: cases where we need to do another pass through this func,
+        #  e.g. the categories are timedelta64s
+        if isna(data).any():
```

Comment:
this is pretty heavy handed, what case is this handling? Why are you handling this at all, should this not just dispatch directly to the categorical to return something that can be dealt with?

Reply:
If there is a `NaT`, that doesn't show up in `codes` ... and I'm just now getting why you suggested using `take`.  Will see if these cases can be unified.